src/main/java/net/kodehawa/mantarobot/commands/utils/polls/Poll.java

```diff
@@ -0,0 +1,287 @@
+    private static final ManagedDatabase db = MantaroData.db();
+    private static final String ztable = "zpoll";
+    private static final String table = "poll";
+
+    // Jackson infuriates me sometimes...
+    @JsonProperty("messageId")
+    private String messageId;
+    @JsonProperty("guildId")
+    private final String guildId;
+    @JsonProperty("channelId")
+    private final String channelId;
+    @JsonProperty("name")
+    private final String name;
+    @JsonProperty("image")
+    private final String image;
+    @JsonProperty("options")
+    private final List<String> options;
+    @JsonProperty("time")
+    private final long time;
+
+    @JsonCreator
+    @ConstructorProperties({"guildId", "channelId", "messageId", "name", "image", "options", "time"})
+    public Poll(String guildId, String channelId, String messageId, String name, String image, List<String> options, long time) {
+        this.guildId = guildId;
+        this.channelId = channelId;
+        this.messageId = messageId;
+        this.name = name;
+        this.image = image;
+        this.options = options;
+        this.time = time;
+    }
+
+    public String guildId() {
+        return guildId;
+    }
+
+    public String channelId() {
+        return channelId;
+    }
+
+    public String messageId() {
+        return messageId;
+    }
+
+    public String name() {
+        return name;
+    }
+
+    public String image() {
+        return image;
+    }
+
+    public List<String> options() {
+        return options;
+    }
+
+    public long time() {
+        return time;
+    }
+
+    public String id() {
+        if (messageId() == null) {
+            throw new IllegalArgumentException("Haven't set message id!");
+        }
+
+        return channelId() + ":" + messageId();
+    }
+
+    @JsonIgnore
+    private void setMessageId(String messageId) {
+        if (messageId() != null) {
+            throw new IllegalArgumentException("Already set message id!");
+        }
+
+        this.messageId = messageId;
+    }
+
+    public void start(SlashContext ctx) {
+        // We might need this sanity checks in case we somehow get a delay on getting the call and the guild has already left.
+        // This has happened in Audio before, so better safe than sorry.
+        if (getGuild() == null)
+            return;
+
+        // Channel somehow doesn't exist anymore? Nowhere to send the message.
+        if (getChannel() == null) {
+            ctx.edit("commands.poll.invalid_channel", EmoteReference.ERROR);
+            return;
+        }
+
+        var at = 0;
+        var user = ctx.getAuthor();
+        var languageContext = ctx.getLanguageContext();
+        var builder = new EmbedBuilder().setAuthor(String.format(languageContext.get("commands.poll.header"), user.getName()), null, user.getAvatarUrl())
+                .setColor(Color.CYAN)
+                .setThumbnail("https://i.imgur.com/7TITtHb.png")
```

Comment:
Might be better to host images ourselves

Reply:
Yea, we should go through all the external hotlinks and do just that.